wsimod/orchestration/model.py

```diff
@@ -718,55 +745,12 @@ def enablePrint(stdout):
                 node.t = date
                 node.monthyear = date.to_period("M")
 
-            # Run FWTW
-            for node in self.nodes_type.get("FWTW", {}).values():
-                node.treat_water()
-
-            # Create demand (gets pushed to sewers)
-            for node in self.nodes_type.get("Demand", {}).values():
-                node.create_demand()
-
-            # Create runoff (impervious gets pushed to sewers, pervious to groundwater)
-            for node in self.nodes_type.get("Land", {}).values():
-                node.run()
-
-            # Infiltrate GW
-            for node in self.nodes_type.get("Groundwater", {}).values():
-                node.infiltrate()
-
-            # Discharge sewers (pushed to other sewers or WWTW)
-            for node in self.nodes_type.get("Sewer", {}).values():
-                node.make_discharge()
-
-            # Foul second so that it can discharge any misconnection
-            for node in self.nodes_type.get("Foul", {}).values():
-                node.make_discharge()
-
-            # Discharge WWTW
-            for node in self.nodes_type.get("WWTW", {}).values():
-                node.calculate_discharge()
-
-            # Discharge GW
-            for node in self.nodes_type.get("Groundwater", {}).values():
-                node.distribute()
-
-            # river
-            for node in self.nodes_type.get("River", {}).values():
-                node.calculate_discharge()
-
-            # Abstract
-            for node in self.nodes_type.get("Reservoir", {}).values():
-                node.make_abstractions()
-
-            for node in self.nodes_type.get("Land", {}).values():
-                node.apply_irrigation()
-
-            for node in self.nodes_type.get("WWTW", {}).values():
-                node.make_discharge()
-
-            # Catchment routing
-            for node in self.nodes_type.get("Catchment", {}).values():
-                node.route()
+            # Iterate over orchestration
+            for timestep_item in self.orchestration:
+                for node_type, function in timestep_item.items():
+                    if node_type in self.nodes_type.keys():
+                        for node in self.nodes_type[node_type].values():
+                            getattr(node, function)()
```

Comment:
This can be done in the same way it was done before:
```suggestion
                    for node in self.nodes_type.get(node_type, {}).values():
                        getattr(node, function)()
```